scripts/west_commands/patch.py

```diff
@@ -190,12 +199,20 @@ def filter_args(self, args):
         topdir = Path(self.topdir)
         manifest_dir = topdir / manifest_path
 
-        if args.patch_base.is_relative_to(_WEST_MANIFEST_DIR):
+        if issubclass(type(args.patch_base), Path):
             args.patch_base = manifest_dir / args.patch_base.relative_to(_WEST_MANIFEST_DIR)
-        if args.patch_yml.is_relative_to(_WEST_MANIFEST_DIR):
+        else:
+            args.patch_base = manifest_dir / args.patch_base
```

Comment:
I don't like this approach of depending on the argument's type to change the behavior.

It's currently expanded (by default) to `Path("WEST_MANIFEST_DIR/zephyr/patches")`, so the relative aspect can still apply.